test/test_sparse.py

```diff
@@ -1527,6 +1527,7 @@ def fn(S):
         for test_dim in test_dims:
             S = self._gen_sparse(sparse_dims, nnz, with_size, dtype, device, coalesced)[0]
             run_tests(S.requires_grad_(True), test_dim)
+            run_tests(S.requires_grad_(True), test_dim, dtype)
```

Comment:
The input dtype is the dtype that the input tensors should have. Here you should add a few dtypes and iterate in the for loop over the product of `test_dims` and those dtypes.